cmd/omniwitness/README.md

```diff
@@ -34,6 +34,36 @@ at https://play.golang.org/p/uWUKLNK6h9v. It is recommended to copy this code
 to your local machine and run from there in order to minimize the risk to the
 private key material.
 
+### Bastion Support
+
+This witness implementation supports synchronous witnessing via the [HTTPS bastion](https://github.com/C2SP/C2SP/blob/main/https-bastion.md) protocol.
+
+To enable this, two flags must be passed to `omniwitness`:
+
+1. `--bastion_addr` is the `host:port` of the bastion host to connect to.
+1. `--bastion_key_path` is the path to a file containing an ed25519 private key in PKCS8 PEM format.
+
+Although the witness key _could_ be reused, it's strongly recommended to use a separate key for this. Such a key can be generated with the following command:
+
+```bash
+openssl genpkey -algorith ed25519 -out ./my_bastion_private_key.pem
+```
+
+The bastion host operator will need to be given the hash of the corresponding public key bytes in order to provision the witness.
```

Comment:
Surely the bastion operator will need to be given the actual public key? I get that the hash will be used in the protocol, but if the keypair is being used to authenticate the witness then surely just a hash of the public part isn't enough. Or I need to go back to crypto school and learn some more...

Reply:
Pretty sure it doesn't :)

The witness creates a self-signed (or not, whatevs) certificate which wraps the "backend" public key and uses it as the client-side cert for the TLS connection to the bastion. The bastion then [accepts the TLS connection iff the hash of the wrapped public key is in the set of accepted backends](https://github.com/FiloSottile/litetlog/blob/main/bastion/bastion.go#L101-L112).

The corresponding "backend" private key is only ever used by TLS machinery.

Reply:
They can, of course, but how is their TLS handshake going to work without the corresponding private key?

Reply:
OK, I think I see how this works now.

1. The bastion operator is given a hash of a public key X
2. A witness operator then creates a cert that contains X, and this cert is signed by the private key for X
3. The bastion operator accepts the connection because X hashes to something it knows, and then verifies that the cert can be verified with X (i.e. was signed with the corresponding private key)

Is that right? I can't see why X can't just be supplied to the bastion operator instead of H(X), but I can see how this can work now at least.

Reply:
More or less, yup :)

In a way, this is not all that different from the HAB fusing in the AW devices - ~ you flash H(X_pub) into the fuses, then "send" firmware which has both a signature by X_sec, and X_pub attached, then you "accept" the firmware if the signature is correct and H(X_pub) matches your fuses.